apm-sniffer/optional-plugins/logger-plugin/src/main/java/org/apache/skywalking/apm/plugin/logger/define/Log4j2LoggerInstrumentation.java

```diff
@@ -0,0 +1,81 @@
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.apm.plugin.logger.define;
+
+import net.bytebuddy.description.method.MethodDescription;
+import net.bytebuddy.matcher.ElementMatcher;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.ConstructorInterceptPoint;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.InstanceMethodsInterceptPoint;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.ClassInstanceMethodsEnhancePluginDefine;
+import org.apache.skywalking.apm.agent.core.plugin.match.ClassMatch;
+import org.apache.skywalking.apm.agent.core.plugin.match.NameMatch;
+import org.apache.skywalking.apm.plugin.logger.ContextConfig;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static net.bytebuddy.matcher.ElementMatchers.named;
+
+public class Log4j2LoggerInstrumentation extends ClassInstanceMethodsEnhancePluginDefine {
+    private static final String ENHANCE_CLASS = "org.apache.logging.log4j.spi.AbstractLogger";
+    private static final String INTERCEPT_CLASS = "org.apache.skywalking.apm.plugin.logger.Log4j2LoggerInterceptor";
+    private static final ContextConfig.LoggerConfig CONFIG = ContextConfig.getInstance().getLog4j2Config();
+
+    @Override
+    protected ClassMatch enhanceClass() {
+        return NameMatch.byName(ENHANCE_CLASS);
+    }
+
+    @Override
+    public ConstructorInterceptPoint[] getConstructorsInterceptPoints() {
+        return new ConstructorInterceptPoint[0];
+    }
+
+    @Override
+    public InstanceMethodsInterceptPoint[] getInstanceMethodsInterceptPoints() {
+        if (CONFIG == null) {
+            return null;
+        }
+        return new InstanceMethodsInterceptPoint[]{
+                new InstanceMethodsInterceptPoint() {
+
+                    @Override
+                    public ElementMatcher<MethodDescription> getMethodsMatcher() {
+                        List<String> levelList = CONFIG.getUpperLevelList(CONFIG.getLevel())
+                                .stream().map(it -> it.name().toLowerCase()).collect(Collectors.toList());
+                        ElementMatcher.Junction<MethodDescription> canInstrumentMethods = named(levelList.get(0));
+                        for (int i = 1; i < levelList.size(); i++) {
+                            canInstrumentMethods = canInstrumentMethods.or(named(levelList.get(i)));
+                        }
+                        return canInstrumentMethods;
```

Comment:
Take the advantage of the array as return value, `InstanceMethodsInterceptPoint[]`.